src/DatePicker.js

```diff
@@ -1,41 +1,28 @@
-import React from 'react';
+import React, { useMemo, useEffect, useRef } from 'react';
 import PropTypes from 'prop-types';
 import TextInput from './TextInput';
 import idgen from './idgen';
 
-class DatePicker extends React.Component {
-  constructor(props) {
-    super(props);
+const DatePicker = props => {
+  const id = useMemo(() => props.id || `datepicker${idgen()}`, [props.id]);
```

Comment:
sorry for the back and forth, but it's best to move id to default props
`id: datepicker${idgen()}` and this is done :)